evm/src/cpu/memio.rs

```diff
@@ -38,14 +41,47 @@ fn eval_packed_load<P: PackedField>(
     for &channel in &lv.mem_channels[4..NUM_GP_CHANNELS - 1] {
         yield_constr.constraint(filter * channel.used);
     }
+
+    // Stack behavior constraints
```

Comment:
Can't we mimic what's been done in #1184 to remove all this code duplication and make it clearer?